src/dali/in_transaction.py

```diff
@@ -36,6 +36,7 @@ def __init__(
         asset: str,
         exchange: str,
         holder: str,
+        # TODO not sure why we aren't using `Keyword` here?
```

Comment:
Could you elaborate? Not sure I understand the comment.

Reply:
The `configuration.Keyword` enum looks like what the input values should be here, curious what this isn't used as the type instead of a raw string.

Reply:
Most of the parameters to instantiate a transaction are strings (they come from REST/CSV sources or from the configuration file), so the design choice has been to pass in strings and then [validate them inside the transaction constructor](https://github.com/eprbell/dali-rp2/blob/main/src/dali/in_transaction.py#L61). 

However it's true that for `transaction_type` we could have used an enum, because `transaction_type` is assigned in the code and not read from external input. I guess I chose string for uniformity with the rest of the parameters.

Note that the `transaction_type` parameter is [validated fully at runtime](https://github.com/eprbell/dali-rp2/blob/main/src/dali/in_transaction.py#L63) and even if we changed transaction_type to enum, we would still need to [validate the value](https://github.com/eprbell/dali-rp2/blob/5f91dd3c8fc9d4548ea9f9112936b97978e4decd/src/dali/in_transaction.py#L26), depending on the actual class type of the transaction (see the [DIRECTION_2_TRANSACTION_TYPE_SET](https://github.com/eprbell/dali-rp2/blob/main/src/dali/configuration.py#L125) dictionary) so not much would change in the code (we would only skip the string check).